### PR TITLE
Port rest of integration tests to Nomad

### PR DESCRIPTION
**Pull request by ghost**
### Which issue does this PR correspond to?
grapl-security/issue-tracker#659

### What changes does this PR make to Grapl? Why?
This gets all the integration tests running (but not passing) in Nomad

### How were these changes tested?
make test-integration

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,6 +154,11 @@ build-test-integration: build
 	$(WITH_LOCAL_GRAPL_ENV) \
 	$(DOCKER_BUILDX_BAKE) --file ./test/docker-compose.integration-tests.yml
 
+.PHONY: push-integration
+push-integration: build-test-integration
```
Comment (twunderlich-grapl) [inline Makefile:158]: Add this so we push up the built integration test containers automatically

File: nomad/local/integration-tests.nomad [hunk 2]
```diff
@@ -40,7 +40,14 @@ variable "redis_endpoint" {
 }
 
 locals {
-  log_level = "DEBUG"
+  log_level            = "DEBUG"
+  local_aws_endpoint =  "http://${attr.unique.network.ip-address}:4566"
```
Comment (twunderlich-grapl) [inline nomad/local/integration-tests.nomad:44]: This is definitely hacky at the moment.

File: nomad/local/integration-tests.nomad [hunk 3]
```diff
@@ -104,5 +111,235 @@ job "integration-tests" {
...
+        AWS_REGION                  = var.aws_region
+        GRAPL_AWS_ENDPOINT          = var.aws_endpoint
+        GRAPL_AWS_ACCESS_KEY_ID     = var.aws_access_key_id
+        GRAPL_AWS_ACCESS_KEY_SECRET = var.aws_access_key_secret
+
+        GRAPL_LOG_LEVEL = local.log_level
+
+        # We don't appear to need this set.
+        # TODO double-check why
+        GRAPL_ANALYZER_MATCHED_SUBGRAPHS_BUCKET = ""
```
Comment (twunderlich-grapl) [inline nomad/local/integration-tests.nomad:206]: @wimax-grapl do you know off the top of your head if these environment variables need to be defined at all? for these tests?

File: pulumi/grapl/__main__.py [hunk 4]
```diff
@@ -170,13 +172,41 @@ def main() -> None:
...
+                lambda inputs: {
+                    k: inputs[k]
+                    for k in inputs.keys()
+                    & {
+                        "aws_region",
+                        "deployment_name",
+                        "aws_access_key_id",
+                        "aws_access_key_secret",
+                        "aws_endpoint",
+                        "redis_endpoint",
```
[the remark this answers is not in the document]
Comment (twunderlich-grapl) [reply pulumi/grapl/__main__.py]: Eventually yes.

File: nomad/local/integration-tests.nomad [hunk 5]
```diff
@@ -104,5 +111,235 @@ job "integration-tests" {
...
+      config {
+        image      = "${var.container_registry}/grapl/analyzerlib-test:latest"
+        entrypoint = ["/bin/bash", "-o", "errexit", "-o", "nounset", "-c"]
+        command    = "cd grapl_analyzerlib && py.test -v -n auto -m 'integration_test'"
+      }
+
+      env {
+        DEPLOYMENT_NAME = var.deployment_name
+        GRAPL_LOG_LEVEL = local.log_level
+        MG_ALPHAS       = "localhost:9080"
```
Comment (wimax-grapl) [inline nomad/local/integration-tests.nomad:151]: I suspect it actually needs to be that `localhost_within_bridge` but if it works it works?

File: nomad/local/integration-tests.nomad (hunk 3 above)
Comment (wimax-grapl) [reply nomad/local/integration-tests.nomad:206]: We probably do need them to be defined; I see it in use inside of Analyzer Executor (not necessarily the test) at:
```
            for exec_hit in self.poll_process(rx=rx, analyzer_name=analyzer_name):
                with self.metric_reporter.histogram_ctx(
                    "analyzer-executor.emit_event.ms",
                    (TagPair("analyzer_name", exec_hit.analyzer_name),),
                ):
                    emit_event(self.analyzer_matched_subgraphs_bucket, s3, exec_hit)
```

basically, emitting to S3 the next step in the pipeline

**Review by wimax-grapl — APPROVED** [review]

File: nomad/local/integration-tests.nomad [hunk 6]
```diff
@@ -201,11 +201,10 @@ job "integration-tests" {
...
 
-        # We don't appear to need this set.
-        # TODO double-check why
-        GRAPL_ANALYZER_MATCHED_SUBGRAPHS_BUCKET = ""
-        GRAPL_ANALYZERS_BUCKET                  = ""
-        GRAPL_MODEL_PLUGINS_BUCKET              = ""
+        # These environment vars need to exist but the values aren't actually exercised
+        GRAPL_ANALYZER_MATCHED_SUBGRAPHS_BUCKET = "NOT_ACTUALLY_EXERCISED_IN_TESTS"
+        GRAPL_ANALYZERS_BUCKET                  = "NOT_ACTUALLY_EXERCISED_IN_TESTS"
+        GRAPL_MODEL_PLUGINS_BUCKET              = "NOT_ACTUALLY_EXERCISED_IN_TESTS"
```
Comment (wimax-grapl) [inline nomad/local/integration-tests.nomad:207]: love it ship it

**Pull request merged**